source/FlareComponent.js

```diff
@@ -49,38 +49,35 @@ export default class FlareComponent extends React.Component
 		return this.canvasRef.current;
 	}
 
-	componentWillReceiveProps(nextProps)
+	componentDidUpdate(prevProps, prevState) 
 	{
-		if (nextProps.isPaused !== this.props.isPaused)
-		{
-			if (
-				!nextProps.isPaused &&
+		if (this.props.isPaused !== prevProps.isPaused) 
+		{		
+			if (!prevProps.isPaused &&
 				this._RuntimeAnimation &&
-				this._AnimationTime === this._RuntimeAnimation._Duration
-			)
+				this._AnimationTime === this._RuntimeAnimation._Duration)
 			{
 				this._AnimationTime = 0.0;
 			}
 			this.startRenderLoop();
 		}
-
-		if (nextProps.file !== this.props.file)
+		if (this.props.file !== prevProps.file)
 		{
-			this.load(nextProps.file);
+			this.load(prevProps.file);
```

Comment:
It looks like this will attempt to load the previously set file. I think we want to load the incoming change (what used to be nextProps, which in this case is the this.props). Similarly below it'll try to initialze to the previous artboardName and controller. 

We might want to make a test (maybe a button in your example) that changes the current file and make sure it's changing to the expected value.

You could use a second .flr file that has a cactus that's a different color or something. I'm worried that the code as it is will not change to the new file.